.github/workflows/publish.yml

```diff
@@ -0,0 +1,160 @@
+name: Publish
+
+on:
+  workflow_dispatch:
+  release:
+    types:
+      - published
+
+jobs:
+  compile_core:
+    name: Compile Core
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: default
+          toolchain: stable
+          target: wasm32-wasi
+          default: true
+
+      - name: Get wasm-opt
+        run: |
+          curl -L https://github.com/WebAssembly/binaryen/releases/download/version_116/binaryen-version_116-x86_64-linux.tar.gz > binaryen.tar.gz
+          tar xvzf binaryen.tar.gz
+          sudo cp binaryen-version_116/bin/wasm-opt /usr/local/bin
+          sudo chmod +x /usr/local/bin/wasm-opt
+
+      - name: Make core
+        run: make core
+
+      - name: Opt core
+        run: wasm-opt --enable-reference-types --enable-bulk-memory --strip -O3 lib/target/wasm32-wasi/release/core.wasm -o lib/target/wasm32-wasi/release/core.wasm
+
+      - name: Upload core binary to artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: engine
+          path: lib/target/wasm32-wasi/release/core.wasm
+
+  compile_cli:
+    name: Compile CLI
+    needs: compile_core
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - name: linux
+            os: ubuntu-latest
+            path: target/x86_64-unknown-linux-gnu/release/extism-py
+            asset_name: extism-py-x86_64-linux-${{ github.event.release.tag_name }}
+            shasum_cmd: sha256sum
+            target: x86_64-unknown-linux-gnu
+          - name: linux-arm64
+            os: ubuntu-latest
+            path: target/aarch64-unknown-linux-gnu/release/extism-py
+            asset_name: extism-py-aarch64-linux-${{ github.event.release.tag_name }}
+            shasum_cmd: sha256sum
+            target: aarch64-unknown-linux-gnu
+          - name: macos
+            os: macos-latest
+            path: target/x86_64-apple-darwin/release/extism-py
+            asset_name: extism-py-x86_64-macos-${{ github.event.release.tag_name }}
+            shasum_cmd: shasum -a 256
+            target: x86_64-apple-darwin
+          - name: macos-arm64
+            os: macos-latest
+            path: target/aarch64-apple-darwin/release/extism-py
+            asset_name: extism-py-aarch64-macos-${{ github.event.release.tag_name }}
+            shasum_cmd: shasum -a 256
+            target: aarch64-apple-darwin
+          # - name: windows
```

Comment:
did you have some issues with windows? not a blocker just curious

Reply:
Yeah it was failing (maybe because of something about location the python library?) but I want to try to figure out what's going on using an actual Windows machine instead of debugging on CI.